packages/tests/server/regression/issue-2996-defined-data.test.tsx

```diff
@@ -95,7 +95,7 @@ test('using `placeholderData`', async () => {
 
   function MyComponent() {
     const { data: trpcData } = proxy.posts.useQuery(undefined, {
-      placeholderData: [],
+      placeholderData: [] as Post[],
```

Comment:
this gets inferred as `never[]` now that we don't enforce TData to extend TQueryData. Not sure if this is okay to change 🤔